src/Analyzers/CSharp/Tests/UseCollectionInitializer/UseCollectionInitializerTests.cs

```diff
@@ -1205,6 +1207,103 @@ public void M()
         obj.Add(""int"", 1);
         obj.Add("" object"", new { X = 1, Y = 2 });
         }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCollectionInitializer)]
+        [WorkItem(47632, "https://github.com/dotnet/roslyn/issues/47632")]
+        public async Task TestWhenReferencedInInitializerLeft()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+using System.Collections.Generic;
+
+class C
+{
+    static void M()
+    {
+        List<object> items = [|new|] List<object>();
+        items[0] = 1;
+        items[items.Count - 1] = 2;
+    }
+}",
+@"
+using System.Collections.Generic;
+
+class C
+{
+    static void M()
+    {
+        List<object> items = [|new|] List<object>
+        {
+            [0] = 1
+        };
+        items[items.Count - 1] = 2;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCollectionInitializer)]
+        [WorkItem(47632, "https://github.com/dotnet/roslyn/issues/47632")]
+        public async Task TestWithIndexerInInitializerLeft()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+using System.Collections.Generic;
+
+class C
+{
+    static void M()
+    {
+        List<object> items = [|new|] List<object>();
+        items[0] = 1;
```

Comment:
Do we also want a no diagnostic test where this line is not present to match the original repro case in the issue being fixed?

Reply:
sure, i can def add that :)